src/main/java/com/innobridge/ethpay/repository/TransactionRepository.java

```diff
@@ -11,4 +11,19 @@
 public interface TransactionRepository extends MongoRepository<Transaction, String> {
     @Query(value = "{ 'senderId' : ?0, 'status' : ?1, 'sourceCurrency' : ?2}", sort = "{ 'createdDate' : -1 }")
     List<Transaction> findBySenderIdAndStatusAndSourceCurrency(String userId, TransactionStatus status, Currency sourceCurrency);
+
+    @Query(value = "{ '$or': [ { 'senderId': ?0 }, { 'receiverId': ?0 } ], 'status': { '$in': ?1 } }", sort = "{ 'createdDate' : -1 }")
+    List<Transaction> findBySenderIdOrReceiverIdAndStatusIn(String userId, List<TransactionStatus> statuses);
```

Comment:
I think there should be clear distinction in these queries, either query by sender OR reciever but not both together, i am assuming these fields are clearly distinct in the collection...

Reply:
The receive and sender are distinct fields in the colleciton.